Add -Wextra to gen/GNUmakefile

msaitoh opened the pull request:
Tested on both FreeBSD and Ubuntu

Some notes:
a) commit a8fe468633d050f3835c8996bcfdddedbeca2313 might not be required.
It hides warning.

b) On Ubuntu, following two warnings are left:
gen.c:4646:7: warning: variable 'error' set but not used [-Wunused-but-set-variable]
                int error, i;
                    ^
gen.c:4676:7: warning: variable 'error' set but not used [-Wunused-but-set-variable]
                int error, i;
                    ^

msaitoh committed:
consitently use flowid with unsigned.

msaitoh committed:
Modify getifunit() for -Wextra

 Change i from int to u_int and check characters up to ifname[0]
in the for loop. No functional change intended.

msaitoh committed:
Change pps_hz from int to u_int for -Wextra

 pps_hz is alwasys > 1, so use unsinged.

msaitoh committed:
Change min_pktsize from int to u_int for -Wextra

msaitoh committed:
Change opt_npkt_sync from int to u_int for -Wextra.

msaitoh committed:
Change rfc2544_ntest and rfc2544_nthtest from int to u_int for -Wextra

msaitoh committed:
Change the loop count in pathhandler() from int to u_int for -Wextra.

msaitoh committed:
Change linebuffer_cursor from int to u_int for -Wextra.

Note that item->w is little hard to change to unsigned, so cast it to u_int.

msaitoh committed:
Change the second argument of genscript_get_item() unsigned for -Wextra

msaitoh committed:
Add -Wextra to CFLAGS.

msaitoh committed:
Change "ret" variable from int to unsigned int for -Wextra.

Note that the return value of xsk_ring_prod__reserve() is unsigned.

ozaki-r commented on gen/GNUmakefile:
```diff
@@ -6,6 +6,7 @@ CFLAGS+=	-I.. -I/usr/local/include -g -DHTDOCS=\"${PREFIX}/share/ipgen/htdocs\"
 CFLAGS+=	-Wall -Wstrict-prototypes -Wmissing-prototypes -Wpointer-arith
 CFLAGS+=	-Wreturn-type -Wswitch # -Wshadow XXX for gen.c
 CFLAGS+=	-Wcast-qual -Wwrite-strings
+CFLAGS+=	-Wextra -Wno-unused-parameter
```
Let's keep warnings or fix them instead of hiding them by `-Wno-unused-parameter` that would hide future issues.

msaitoh committed:
Sprinkle __unused for signal handlers. No functional change.

 - For -Wextra.
 - Those four functions don't use signal number.

msaitoh committed:
Sprinkle __unused for some itemlist callbacks. No functional change.

 - For -Wextra.
 - Those functions are registered with REG(xxx, fname, NULL), so the
   refptr is not used.

msaitoh committed:
Sprinkle __unused for some itemlist callbacks. No functional change.

msaitoh committed:
Sprinkle __unused for some libevent callbacks. No functional change.

msaitoh committed:
Mark __unused to control_thread_main()'s argument. No functional change.

Other all *_thread_main() functions' argument are void *, so
keep control_thread_main()'s argument as it is though it can be changed
from void * to void.

msaitoh committed:
Sprinkle __testused(__unused) for some callbacks. No functional change.

 - For -Wextra.
 - Some warnings are not fixed.

msaitoh committed:
Define __unused if it's not defined.

msaitoh committed:
Mark __debugused(__unused) to item_dump(). No functional change.

msaitoh committed:
Sprinkle __unused for urlhandler's arguments. No functional change.

msaitoh committed:
Sprinkle __debugused(__unused) for some functions. No functional change.

msaitoh commented:
-Wno-unused-parameter was removed.
Note that some warnings are not fixed.

ozaki-r commented on gen/arpresolv.c:
```diff
@@ -97,6 +97,12 @@ static int getifinfo(const char *, int *, uint8_t *);
 	} while (/* CONSTCOND */ 0)
 
+#ifdef STANDALONE_TEST
+#define	__diagused
```
`__diagused` is usually for `DIAGNOSTIC` so using it for `STANDALONE_TEST` is confusable. How about `__testused` or something?

msaitoh replied:
OK. I'll modify it.

msaitoh replied:
done.

ozaki-r commented on gen/arpresolv_linux.c:
```diff
@@ -88,6 +89,12 @@ struct recvnd_arg {
 static unsigned int buflen = BUFSIZE;
 
+#ifdef DEBUG
+#define __diagused
```
Ditto. In this case `__debugused` is appropriate.

msaitoh replied:
I'll do it.

msaitoh replied:
done

ozaki-r commented:
Also, please update the title of the PR.

msaitoh commented:
Done.

ozaki-r reviewed (commented):
The diffs look ok. Can you clean up (squash) commits?

msaitoh commented:
Done.
Is it enough?

ozaki-r reviewed (approved):
Yes, thank you for tidy-up!

ozaki-r commented:
The self-test passed.

```
ozaki-r@fedora ~/g/i/gen (w-extra)
        > sudo bash selftest.bash
[sudo] password for ozaki-r: 
[0] Testing packet generation benchmark (-X)...passed
[1] Testing packet generation benchmark (-XX)...passed
[2] Testing packet generation benchmark (-XXX)...passed
[3] Testing send/recv test...passed
[4] Testing send/recv test (full-duplex)...passed
[5] Testing send/recv test with IPv4...passed
[6] Testing send/recv test with IPv6...passed
[7] Testing send/recv test with IPv4 (TCP)...passed
[8] Testing send/recv test with IPv6 (TCP)...passed
[9] Testing send/recv test with IPv4 (TCP, fragment)...passed
[10] Testing send/recv test with IPv6 (TCP, fragment)...passed
[11] Testing send/recv test with IPv4 (range)...passed
[12] Testing send/recv test with IPv6 (range)...passed
[13] Testing send/recv test with IPv4 (allnet)...passed
[14] Testing send/recv test with IPv6 (allnet)...passed
[15] Testing RFC 2544 test with IPv4, take a while...passed
[16] Testing RFC 2544 test with IPv6, take a while...passed
[17] Testing send/recv test with VLAN...passed
```
